common.mk

```diff
@@ -1,9 +1,96 @@
-MKFILE_PATH := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
-NPROCS = $(shell sysctl hw.ncpu  | grep -o '[0-9]\+')
-MAKEFLAGS += -j$(NPROCS)
-
-export BIN_DIR := $(MKFILE_PATH)bin
-export DEV_DATA_DIR := $(MKFILE_PATH)data
-export DEV_PROC_DIR := $(DEV_DATA_DIR)/procs
-export DEV_COMMUNITY_DIR := $(DEV_DATA_DIR)/communities
-export APPS_DIR := $(MKFILE_PATH)apps
\ No newline at end of file
+MIN_MAKE_VERSION	=	4.0.0
```

Comment:
i'm going to be honest, don't really understand a bunch of this but trust that you tested it